lxd/apparmor/instance_qemu.go

```diff
@@ -41,8 +41,8 @@ profile "{{ .name }}" flags=(attach_disconnected,mediate_deleted) {
   /{,usr/}bin/qemu-system-*                 mrix,
   /usr/share/qemu/**                        kr,
   /usr/share/seabios/**                     kr,
-  owner @{PROC}/@{pid}/cpuset               r,
```

Comment:
What does removing the `owner` part mean (in theory)?

Reply:
according to the AppArmor man:
```
       owner
           Specifies that the task must have the same euid/fsuid as the object being referenced
           by the permission check.
```

So, this rule was applied *only* to the files which are owned by the user who runs Qemu, but after removing it we allow Qemu to access any other files too. To be more precise, AppArmor won't disallow access to other files, at the same time generic Linux permission checks are still applied.

Reply:
So is `@{PROC}/@{pid}/cpuset` normally owned by the same user as qemu is run as? But for some reason apparmor is deny access?

Is PROC in this case the `/proc`?

So we are only allowing it access to files inside its own pid anyway?

This seems fine, there are a couple of other uses of `@{PROC}/@{pid}/cpuset` without `owner` in LXD already.

Reply:
>So is @{PROC}/@{pid}/cpuset normally owned by the same user as qemu is run as? But for some reason apparmor is deny access?

Absolutely.

>Is PROC in this case the /proc?

Yes.

>So we are only allowing it access to files inside its own pid anyway?

with `owner` - yes, without `owner` we allow access to the files for other pids too. But, of course, standard Linux permission checks are still applied.